mne/beamformer/_compute_beamformer.py

```diff
@@ -291,7 +308,7 @@ def _compute_beamformer(G, Cm, reg, n_orient, weight_norm, pick_ori,
                     assert Ck.shape[1:] == (3, 3)
                     # Invert for all dipoles simultaneously using matrix
                     # inversion.
-                    norm = _sym_inv(Ck, reduce_rank)
+                    norm = _sym_inv(Ck, reduce_rank=False)
```

Comment:
Leaving this as `reduce_rank=reduce_rank` fixes two out of the three errors. I think it makes sense to reduce in both places. During inversion you don't want to invert three components, just two in the case of `reduce_rank`.

Reply:
(If we were to make the leadfield actually have dimensionality 2 instead of 3, we could have `reduce_rank=False` here and maybe simplify things. But then we'd have to keep track of the rotation / singular vectors, and rotate back after finishing the computations. Either way should be equivalent, and maybe the 2-dim way a bit faster, not sure which way you want to go...)

Reply:
Sorry I mean `n_orient=2` instead of `n_orient=3` in `reduce_rank=True` mode (not "dimensionality")

Reply:
Hummm, I see what you mean and I can totally see that this would work. It might be unexpected beamformer behaviour for the user though. Ultimately, you reduced the rank of the leadfield already at this point ... 
But on the other hand, it basically is just a "pinv" alternative, right?

Reply:
Well, having `reduce_rank` also reducing the number of orientations could be a way to handle things as well (I think FieldTrip for example supports both, with or without backprojection). It would indeed add a bit of tracking though ... And I am not sure how attractive this option would be in the real world. I find it intuitive (to reduce the output to 2 orientations, in case of a vector beamformer) but probably a lot of our orientation selection code might break?

Reply:
> But on the other hand, it basically is just a "pinv" alternative, right?

Yeah basically you need to pass `reduce_rank` to the inversion functions to make sure they don't blow something up that they shouldn't. If I use a much smaller `tol` in the effective `pinv` calculation it also works, but it seems safer just to explicitly reduce it

Reply:
Having tried now to quickly implement the lower-dimensional solution, I don't think it's worth it. We also would need to transform `Ck` to the lower-dimensional space of size `2`, do everything there, and then project back to the space of size `3`. I doubt the speed gains are meaningful and the code would require more variables to be passed around.

Reply:
Yeah that was my fear as well. So let's stick with `sym_inv(data, reduce_rank=True)`